### ISSUE TITLE
NPE in TextReporter.logResult

### ISSUE THREAD

**Issue by rrozenshteyn-jha** [cbeust/testng#2633]
### TestNG Version
> Note: only the latest version is supported
7.4.0
### Expected behavior
Text report is printed and Java process running TestNG exits normally.

### Actual behavior
The following NPE is thrown and the Java process running TestNG exits with exit code -1 and no other reports are produced.

Exception in thread "main" java.lang.NullPointerException
     at org.testng.internal.Utils$$Lambda$486/0x0000000015601220.apply(Unknown Source)
     at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
     at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:948)
     at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:497)
     at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:487)
     at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:563)
     at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:265)
     at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:517)
     at org.testng.internal.Utils.extractParameterTypes(Utils.java:612)
     at org.testng.reporters.TextReporter.logResult(TextReporter.java:140)
     at org.testng.reporters.TextReporter.logResults(TextReporter.java:74)
     at org.testng.reporters.TextReporter.onFinish(TextReporter.java:34)
     at org.testng.TestRunner.fireEvent(TestRunner.java:923)
     at org.testng.TestRunner.afterRun(TestRunner.java:887)
     at org.testng.TestRunner.run(TestRunner.java:599)
     at org.testng.SuiteRunner.runTest(SuiteRunner.java:377)
     at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:371)
     at org.testng.SuiteRunner.privateRun(SuiteRunner.java:332)
     at org.testng.SuiteRunner.run(SuiteRunner.java:276)
     at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
     at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
     at org.testng.TestNG.runSuitesSequentially(TestNG.java:1212)
     at org.testng.TestNG.runSuitesLocally(TestNG.java:1134)
     at org.testng.TestNG.runSuites(TestNG.java:1063)
     at org.testng.TestNG.run(TestNG.java:1031)
     at org.testng.TestNG.privateMain(TestNG.java:1365)
     at org.testng.TestNG.main(TestNG.java:1329)

### Is the issue reproducible on runner?

- [x] Shell
- [ ] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
> Please, share the test case (as small as possible) which shows the issue
Attached zip file contains the test case which works with testng-7.0.0.jar but does not work with testng-7.4.0.jar.

#### Works:
javac -cp ./testng-7.0.0.jar TextReporterNpeTest.java
java -cp ./testng-7.0.0.jar;./jcommander-1.78.jar;. org.testng.TestNG testng.xml

#### Does not work:
javac -cp ./testng-7.4.0.jar TextReporterNpeTest.java
java -cp ./testng-7.4.0.jar;./jcommander-1.78.jar;. org.testng.TestNG testng.xml

[TestNG_NPE.zip](https://github.com/cbeust/testng/files/7125151/TestNG_NPE.zip)

**Comment by juherr** [comment]
```java
import static org.testng.Assert.*;

import org.testng.annotations.Test;
import org.testng.annotations.DataProvider;

public class TextReporterNpeTest
{
	@Test(dataProvider = "dp")
	public void f(Integer n, String s)
	{
		// test itself doesn't matter
		// what matters is a null parameter value
		assertNotNull(n);
		assertNotNull(s);
	}

	@DataProvider
	public Object[][] dp()
	{
		return new Object[][]
				{
					{ 1, "a" },
					{ 2, "b" },
					{ 3, null }
				};
	}
}
```

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd" >

<suite name="All" verbose="2" thread-count="8" parallel="methods" time-out="180000">
	<test name="Unit Tests">
		<classes>
			<class name="TextReporterNpeTest" />
		</classes>
	</test>
</suite> 
```

**Comment by krmahadevan** [comment]
This is fixed in the upcoming release and should be available when TestNG `7.5.0` has been released.
